### PR TITLE
feat: Add ingestor internal schemas

### PR DESCRIPTION
**Pull request by mattvryan**
This PR adds new schemas that represent the internal messages passed between ingestion stages, and makes it easier to use the validator for these schemas in unit tests.  Note these aren't meant to be used to validate messages in production.

### CODE REVIEW

**Review by jdelbick — APPROVED** [review]
no strong opinions here on the refactoring. i will approve in case it is blocking work but i would recommend waiting for input from Dan :)

File: src/schema-validator.js [hunk 1]
```diff
@@ -14,68 +14,73 @@ import { dirname, join } from 'path';
 import { fileURLToPath } from 'url';
 import { readFile, readdir } from 'fs/promises';
 
-const SCHEMA_INGESTION_REQUEST = 'ingestion-request';
+export const SCHEMA_INGESTION_REQUEST = 'ingestion-request';
+
+const schemas = {};
+
+export async function getSchema(name, dir) {
```
Comment (klcodanr) [inline src/schema-validator.js:21]: Why move this to an exported function instead of a method?

File: src/schema-validator.js [hunk 2]
```diff
@@ -14,68 +14,73 @@ import { dirname, join } from 'path';
...
-   * @param {any} ingestionRequest
+   * @param {any} request
+   * @param {string} schemaName
    * @param {Array<string>} additionalRequiredData
    */
-  async validateIngestionRequest(ingestionRequest, additionalRequiredData) {
-    const schema = await this.#getSchema(SCHEMA_INGESTION_REQUEST);
-    const result = validate(ingestionRequest, schema, {
+  // eslint-disable-next-line class-methods-use-this
+  async validateRequest(request, schemaName, additionalRequiredData) {
```
Comment (klcodanr) [inline src/schema-validator.js:57]: Should this maybe be "validateObject"? Since the thing being validated may not be a request?

**Review by klcodanr — CHANGES_REQUESTED** [review]

**Review by klcodanr — APPROVED** [review]
LGTM

**Pull request merged**